Agents.md

```diff
@@ -0,0 +1,263 @@
+1. Require internal RFC approval
+2. May need cross-platform alignment (Android, Browser, React Native SDKs)
+3. Must consider backwards compatibility
+
+## SDK Philosophy
+
+From `ZEN.md` - these principles guide all development:
+
+1. **Zero crashes caused by SDK code** - Prefer making the SDK non-operational over throwing exceptions
+2. **Small footprint** - Minimize runtime performance impact, library size, and network load
+3. **Stability** - Avoid breaking changes; minor updates must be transparent
+4. **Compatibility** - Support iOS 12.0+, both Swift and Objective-C
+
+Agents must assume the SDK is used in thousands of apps in production. Any change that may alter behavior must be treated as a potential breaking change.
+
+## Code Structure
+
+```
+dd-sdk-ios/
+├── DatadogInternal/     # Shared protocols, types, utilities
+│   ├── Sources/
+│   └── Tests/
+├── DatadogCore/         # Core SDK: initialization, pipeline, networking
+│   ├── Sources/
+│   └── Tests/
+├── DatadogLogs/         # Logging feature
+├── DatadogTrace/        # Local and Distributed tracing
+├── DatadogRUM/          # Real User Monitoring
+├── DatadogSessionReplay/# Session Replay
+├── DatadogCrashReporting/ # Crash Reporting
+├── DatadogWebViewTracking/ # Connecting RUM sessions from mobile apps to RUM sessions hapening within WebViews
+├── DatadogFlags/        # Feature flags
+├── TestUtilities/       # Shared test mocks and helpers
+├── IntegrationTests/    # UI and integration tests
+├── BenchmarkTests/      # Performance benchmarks
+├── E2ETests/            # End-to-end tests
+└── tools/               # Build, lint, and code generation tools
+```
+
+## Testing
+
+### Test Conventions
+- **Follow existing patterns** - Look at sibling test files for conventions
+- Use `TestUtilities` for mocks and helpers
+- Mock naming: `static func mockAny()` for any value, descriptive names for specific scenarios
+- Use `DatadogCoreProxy` for integration testing features
+- Do not test Apple frameworks
+- Do not test purely generated code
+- Do not mock DatadogCore incorrectly (use provided helpers)
+
+### Running Tests
+```bash
+# Unit tests for a specific scheme
+make test-ios SCHEME="DatadogCore iOS"
+make test-ios SCHEME="DatadogInternal iOS"
+
+# All iOS unit tests
+make test-ios-all
+
+# UI/Integration tests
+make ui-test TEST_PLAN="Default"
+
+# Session Replay snapshot tests
+make sr-snapshot-test
+```
+
+## Linting
+
+The project uses SwiftLint with custom rules:
+
+```bash
+# Run linter
+./tools/lint/run-linter.sh
+
+# Auto-fix violations
+./tools/lint/run-linter.sh --fix
+```
+
+### Key Lint Rules
+- **TODO comments must include JIRA reference**: `// TODO: RUM-123 description`
+- **No `UIApplication.shared`**: Use `UIApplication.managedShared` instead
+- **No `URLRequest.allHTTPHeaderFields`**: Use `URLRequest.value(forHTTPHeaderField:)` 
+- **Avoid Required Reason API conflicts**: Don't use names that conflict with Apple's privacy APIs
```

Comment:
question/ Why do we need to list key lint rules as part of the agent instructions? 🤔 From a workflow-philosophy standpoint, shouldn’t it be sufficient for the agent to run our linter (`make lint`) and interpret its output to determine whether the code passes? What’s the value in duplicating those rules here?